src/linker/Linker/TypeNameResolver.cs

```diff
@@ -31,13 +31,45 @@ public TypeReference ResolveTypeName (string typeNameString)
 				return ResolveTypeName (null, assemblyQualifiedTypeName);
 			}
 
-			foreach (var assemblyDefinition in _context.GetReferencedAssemblies ()) {
-				var foundType = ResolveTypeName (assemblyDefinition, parsedTypeName);
-				if (foundType != null)
-					return foundType;
+			if (!isCallFromGetType) {
+				_context.LogWarning ($"Type name strings used for dynamically accessing a type must be assembly qualified.",
```

Comment:
I don't follow the message, it says "must be assembly qualified" but it's only a warning?

Reply:
Thanks! Changed the phrasing here. I've made this a warning since we previously used to go through all loaded assemblies searching for the referenced type (which was wrong); so there could be users who will now have breaking apps with no good way to figure out what is going on. That being said -- users shouldn't lean on the linker behavior to pull a type out of nowhere, and the documentation already specifies that type names should be fully qualified, otherwise `Type.GetType` will only search for it in the calling object's assembly and SPCL (https://docs.microsoft.com/en-us/dotnet/api/system.type.gettype?view=net-5.0)... so I'm also happy to not warn here.